torch/csrc/jit/export.cpp

```diff
@@ -618,34 +618,75 @@ class ScriptModuleSerializer {
 // 1. Several tests that depend on the serialization format details.
 // 2. The emit module hook (since combining the old export and new import code
 //    is going to cause jitter)
-class ScriptModuleSerializer2 : public ScriptModuleSerializer {
+class ScriptModuleSerializer2 {
  public:
   ScriptModuleSerializer2(const std::string& filename)
-      : ScriptModuleSerializer(filename) {}
-
-  ScriptModuleSerializer2(std::ostream* ofs) : ScriptModuleSerializer(ofs) {}
+      : writer_(filename.c_str()) {}
+
+  ScriptModuleSerializer2(std::ostream* ofs) : ofs_(), writer_(ofs) {}
+
+  void writeArchive(const std::string& archive_name, const IValue& value) {
+    std::vector<char> data;
+    Pickler data_pickle(
```

Comment:
It looks like `jit::pickle(...)` would be fine here instead of the manual use of the `Pickler`

Reply:
Right after this it uses `data_pickle.tensorData` which is not exposed by those functions.